timeout in shm_ib/test_ucp_peer_failure_keepalive.kill_receiver/0 with valgrind

https://dev.azure.com/ucfconsort/0b36e3f0-8ab9-4a48-b68b-4b2350e02c88/_apis/build/builds/14924/logs/303
```
2021-03-18T08:12:18.7628107Z [ RUN      ] shm_ib/test_ucp_peer_failure_keepalive.kill_receiver/0 <shm,ib/tag>
2021-03-18T08:27:24.5901093Z /scrap/azure/agent-03/AZP_WORKSPACE/2/s/contrib/../test/gtest/common/test_helpers.cc:54: Failure
2021-03-18T08:27:24.5902830Z Failed
2021-03-18T08:27:24.5903658Z Connection timed out - abort testing
2021-03-18T08:27:24.6097467Z [swx-rdmz-ucx-new-01:7111 :0:7111] Caught signal 6 (Aborted: tkill(2) or tgkill(2))
2021-03-18T08:28:45.0793591Z ==== backtrace (tid:   7111) ====
2021-03-18T08:28:45.0802843Z  0 0x00000000000e2d45 __GI___sched_yield()  :0
2021-03-18T08:28:45.0815952Z  1 0x000000000090a5dc ucp_test::progress()  /scrap/azure/agent-03/AZP_WORKSPACE/2/s/contrib/../test/gtest/ucp/ucp_test.cc:156
2021-03-18T08:28:45.0819022Z  2 0x000000000090a5dc ucp_test::request_process()  /scrap/azure/agent-03/AZP_WORKSPACE/2/s/contrib/../test/gtest/ucp/ucp_test.cc:217
2021-03-18T08:28:45.0820591Z  3 0x000000000090a966 ucp_test::request_wait()  /scrap/azure/agent-03/AZP_WORKSPACE/2/s/contrib/../test/gtest/ucp/ucp_test.cc:240
2021-03-18T08:28:45.0822064Z  4 0x000000000090a9d8 ucp_test::cleanup()  /scrap/azure/agent-03/AZP_WORKSPACE/2/s/contrib/../test/gtest/ucp/ucp_test.cc:78
2021-03-18T08:28:45.0829882Z  5 0x00000000005e4190 ucs::test_base::TearDownProxy()  /scrap/azure/agent-03/AZP_WORKSPACE/2/s/contrib/../test/gtest/common/test.cc:333
2021-03-18T08:28:45.0831769Z  6 0x00000000005c7f69 HandleSehExceptionsInMethodIfSupported<testing::Test, void>()  /scrap/azure/agent-03/AZP_WORKSPACE/2/s/contrib/../test/gtest/common/gtest-all.cc:3562
2021-03-18T08:28:45.0833396Z  7 0x00000000005bc17c testing::Test::Run()  /scrap/azure/agent-03/AZP_WORKSPACE/2/s/contrib/../test/gtest/common/gtest-all.cc:3643
2021-03-18T08:28:45.0834921Z  8 0x00000000005bc285 testing::TestInfo::Run()  /scrap/azure/agent-03/AZP_WORKSPACE/2/s/contrib/../test/gtest/common/gtest-all.cc:3812
2021-03-18T08:28:45.0836422Z  9 0x00000000005bc3ef testing::TestCase::Run()  /scrap/azure/agent-03/AZP_WORKSPACE/2/s/contrib/../test/gtest/common/gtest-all.cc:3930
2021-03-18T08:28:45.0838160Z 10 0x00000000005c0b9d testing::internal::UnitTestImpl::RunAllTests()  /scrap/azure/agent-03/AZP_WORKSPACE/2/s/contrib/../test/gtest/common/gtest-all.cc:5808
2021-03-18T08:28:45.0840192Z 11 0x00000000005c0e80 testing::internal::UnitTestImpl::RunAllTests()  /scrap/azure/agent-03/AZP_WORKSPACE/2/s/contrib/../test/gtest/common/gtest-all.cc:5725
2021-03-18T08:28:45.0841800Z 12 0x000000000055511d RUN_ALL_TESTS()  /scrap/azure/agent-03/AZP_WORKSPACE/2/s/contrib/../test/gtest/common/gtest.h:20059
2021-03-18T08:28:45.0843208Z 13 0x000000000055511d main()  /scrap/azure/agent-03/AZP_WORKSPACE/2/s/contrib/../test/gtest/common/main.cc:105
2021-03-18T08:28:45.0844155Z 14 0x00000000000223d5 __libc_start_main()  ???:0
2021-03-18T08:28:45.0844868Z 15 0x00000000005a650d _start()  ???:0
2021-03-18T08:28:45.0846641Z =================================
2021-03-18T08:28:45.0977055Z [swx-rdmz-ucx-new-01:7111 :0:7111] Process frozen...
...
2021-03-18T10:43:25.7983341Z make: *** [test_valgrind] Terminated
2021-03-18T10:43:26.2768894Z ##[error]Bash exited with code '124'.
```

--- a/src/ucp/rma/flush.c
+++ b/src/ucp/rma/flush.c
@@ -519,7 +519,7 @@ static unsigned ucp_worker_flush_progress(void *arg)
         (&next_ep->ep_list != &worker->all_eps)) {
         /* Some endpoints are not flushed yet. Take the endpoint from the list
          * and start flush operation on it. */
-        ep = ucp_worker_flush_req_set_next_ep(req, 1, &next_ep->ep_list);
+        ep = ucp_worker_flush_req_set_next_ep(req, 1, next_ep->ep_list.next);
         if (ep == NULL) {
             goto out;
         }

